src/components/CellsContainer/CellsContainer.svelte

```diff
@@ -0,0 +1,201 @@
+<script>
+  import { onMount, onDestroy } from 'svelte';
+  import { select } from 'd3';
+  import { createSvg, getId, createRectsData, setAttributes } from './helpers';
+
+  export let count = 2;
+  export let size = 100;
+  export let gap = 1;
+  export let getAdditionalRectProps;
+  export let tooltipBackground = '#4a5568';
+  export let tooltipColor = 'MintCream';
+  export let color = 'SteelBlue';
+  export let hoverColor = 'IndianRed';
+  export let margin = { top: 20, left: 20, right: 20, bottom: 20 };
+
+  let containerRef = null;
+  let svg = null;
+  let g = null;
+  let data = [...Array(count)];
+  let containerWidth = 0;
+  let height = 0;
+  let innerWidth = 0;
+
+  $: {
+    if (containerWidth) {
+      innerWidth = containerWidth - margin.left - margin.right;
+
+      const rectSize = size + gap * 2;
+      const colsAmount = Math.floor(innerWidth / rectSize);
+      const rowsAmount = Math.ceil(count / colsAmount);
+
+      height = rectSize * rowsAmount + margin.top + margin.bottom;
+
+      data = createRectsData({
+        containerWidth: innerWidth,
+        count,
+        size,
+        gap,
+        getAdditionalRectProps,
+        color,
+        hoverColor
+      });
+
+      updateSvgElementsSizes();
```

Comment:
you are not using this function anywhere else, so you probably can just type the code here. Or combine all code, `updateRects` as well, into one function
It's a bit strange that part of code is here and other one is in functions
Again, if it is possible to extract all calculations into a module. that would be the best option